### PR TITLE
fix: Run CI checks only once on PRs, not twice

### PR DESCRIPTION
**Pull request by alamb**
I have noticed that the github checks are run twice when I open a PR. Not only is this wasteful of resources (which I suppose we aren't directly paying for now), but I think it means the tests are likely to take longer (as now you have twice the chance for each test to get slowed down by whatever github machinations are going on)

For example: https://github.com/influxdata/influxdb_iox/pull/543

Screen Shot 2020-12-10 at 11 38 06 AM

This PR changes github actions to run on push rather than on PR -- as any commit that is in a PR was also by definition pushed to the repo. LOL

### CODE REVIEW

**Comment by domodwyer** [comment]
This came up at the time - I think @e-dard wanted to keep it building for pushes & PRs?

**Comment by e-dard** [comment]
Yeah it is indeed annoying @alamb. I hoped it would be possible to separate the set of jobs that run on a PR from those that run on a branch but I don't know if it is. 

Personally I like running CI on a branch as it allows you to check things are correct before you open a PR, but I agree it's annoying now to see the duplication on a PR. It's like Github is treating a PR as both a PR and branch push, which is what it is (Github creates a new branch). I'm in favour of nuking the duplication if it's getting annoying

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # Cargo's build artefacts are cached to reduce the build time, see
 # https://github.com/actions/cache for more info.
 
-on: [push, pull_request]
+on: [push]
```
Comment (e-dard) [inline .github/workflows/rust.yml:18]: Is this going to do what we want? If this still works fine for pull requests from our branches and forks I guess my question is - what's the point of having a `on: [pull_request]` option at all? 😕

Comment (e-dard) [reply .github/workflows/rust.yml:18]: I mean it certainly looks from the checks that this does the right thing but I'm confused about the meaning of the `pull_request` option in general. Will this work fine for forks?

Comment (alamb) [reply .github/workflows/rust.yml:18]: I think the distinction is `push` means the the checks get run on commits that are *not* part of PRs whereas `pull_request` means run the checks only if the commits are part of PRs. 

I think the `push` is a reference to the github push webhook: https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#push

And I think  `pull_request` is https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request

**Comment by domodwyer** [comment]
I dunno if this is important or not, but I just noticed that #536 does not have both `push` & `pull_request` events:

image

I had just clicked "Update Branch" so maybe this doesn't count as a push?

**Comment by alamb** [comment]
🤔 I can do some more testing maybe and see what happens

**Comment by alamb** [comment]
FWIW when I clicked "update branch" on https://github.com/influxdata/influxdb_iox/pull/540 then all 8 tests seem to be running 🤔 

Screen Shot 2020-12-15 at 8 43 24 AM

Maybe updating on a branch that is not in the influxdata/influxdb_iox repo doesn't trigger the CI push?

**Comment by carols10cents** [comment]
What if we only keep `pull_request` instead of `push`? And then use the draft PR feature to check before submitting for review?

**Comment by alamb** [comment]
> What if we only keep pull_request instead of push? And then use the draft PR feature to check before submitting for review?

That would work for me -- what do you think @e-dard ?

File: .github/workflows/rust.yml (hunk 1 above)
Comment (alamb) [inline .github/workflows/rust.yml:18]:
```suggestion
on: [pull_request]
```

**Comment by alamb** [comment]
I changed the PR to only run on `pull_request` -- and we'll see how that works

**Review by e-dard — APPROVED** [review]
only running on pull request seems like a good idea to me 👍

**Comment by alamb** [comment]
I will merge this when green

**Pull request merged**